dependency-injection/DependencyInjection/Controller.cs

```diff
@@ -3,12 +3,28 @@
 namespace DependencyInjection;
 
 [ApiController]
-public class Controller(Singleton _singleton)
+public class Controller(IServiceProvider _serviceProvider, Singleton _singleton)
 {
     [HttpPost]
     [Route("stuff")]
     public void DoStuff()
```

Comment:
document this dependency injection usage in readme as well. provide a separate section for controllers and suggest `[FromServices]` and `[FromKeyedServices]` usages.